src/libs/fileDownload/DownloadUtils.ts

```diff
@@ -0,0 +1,67 @@
+import * as ApiUtils from '@libs/ApiUtils';
+import tryResolveUrlFromApiRoot from '@libs/tryResolveUrlFromApiRoot';
+import * as Link from '@userActions/Link';
+import CONST from '@src/CONST';
+import * as FileUtils from './FileUtils';
+import type {FileDownload} from './types';
+
+/**
+ * The function downloads an attachment on web/desktop platforms.
+ */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const fetchFileDownload: FileDownload = (url, fileName, successMessage = '', shouldOpenExternalLink = false, formData = undefined, requestType = 'get', onDownloadFailed?: () => void) => {
+    const resolvedUrl = tryResolveUrlFromApiRoot(url);
+    if (
+        // we have two file download cases that we should allow 1. dowloading attachments 2. downloading Expensify package for Sage Intacct
+        shouldOpenExternalLink ||
+        (!resolvedUrl.startsWith(ApiUtils.getApiRoot()) &&
+            !CONST.ATTACHMENT_LOCAL_URL_PREFIX.some((prefix) => resolvedUrl.startsWith(prefix)) &&
+            url !== CONST.EXPENSIFY_PACKAGE_FOR_SAGE_INTACCT)
+    ) {
```

Comment:
this condition gave me a panic attack 😅 
I'd suggest extracting 1 or 2 parts of it to give them reasonable names, and then simplifying this.